oak/server/http_client_node.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/server/http_client_node.h"
+
+#include "absl/memory/memory.h"
+#include "asylo/util/logging.h"
+#include "grpcpp/create_channel.h"
+#include "oak/common/app_config.h"
+#include "oak/proto/grpc_encap.pb.h"
+#include "oak/proto/http_client.pb.h"
+
+namespace oak {
+
+HttpClientNode::HttpClientNode(const std::string& name, const std::string& url,
+                               const std::string& method)
+    : NodeThread(name), url_(url), method_(method) {}
+
+void HttpClientNode::Run() {
+  Handle request_handle = FindChannel(kHttpClientNodeRequestPortName);
+  Handle response_handle = FindChannel(kHttpClientNodeResponsePortName);
+  MessageChannelReadHalf* request_channel = BorrowReadChannel(request_handle);
```

Comment:
(drive-by): Can I suggest that you just have a single pre-configured channel, and include the handle for an ephemeral response channel on each inbound request?  That would be more consistent with how the gRPC channel works now, and with how the storage channel would work after #334.  

(We're also likely to move from statically-configured nodes and channels to dynamically-created nodes, which will work better with a single inbound channel to an HTTP client pseudo-Node)